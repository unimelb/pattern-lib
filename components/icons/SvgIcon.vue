<template>
  <svg
    :width="width"
    :height="height"
    :aria-label="name"
    focusable="false"
    role="img">
    <use :xlink:href="'#icon-' + name" />
    <title>{{ name | capitalize }}</title>
    <slot />
  </svg>
</template>

<script>
// TODO: investigate refactoring to import svgs dynamically using <path> or base64 in CSS
export default {
  name: 'SvgIcon',
  filters: {
    capitalize(value) {
      const startFromCharIndex = 0;

      let capitalisation = value;
      if (!capitalisation) {
        return '';
      }
      capitalisation = capitalisation.toString();

      return capitalisation.charAt(startFromCharIndex).toUpperCase() + capitalisation.slice(1);
    },
  },
  props: {
    name: {
      type: String,
      default: '',
    },
    width: {
      type: [String, Number],
      default: '0',
    },
    height: {
      type: [String, Number],
      default: '0',
    },
  },
};
</script>
