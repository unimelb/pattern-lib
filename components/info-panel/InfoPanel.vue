<template>
  <div class="info-panel">
    <SvgIcon
      v-if="iconName"
      :name="iconName"
      :aria-label="iconName"
      class="info-panel__icon"
      width="20px"
      height="20px" />
    <div class="info-panel__inner">
      <h4 class="title--xxs info-panel__title">
        {{ title }}
      </h4>
      <slot />
    </div>
  </div>
</template>

<script>
import SvgIcon from '../icons/SvgIcon.vue';

export default {
  name: 'InfoPanel',
  components: { SvgIcon },
  props: {
    iconName: {
      type: String,
      default: '',
    },
    title: {
      type: String,
      default: '',
      required: true,
    },
  },
};
</script>
