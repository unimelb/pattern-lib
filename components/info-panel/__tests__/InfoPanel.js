import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import InfoPanel from '../InfoPanel.vue';
import SvgIcon from '../../icons/SvgIcon.vue';


expect.extend(toHaveNoViolations);

describe('InfoPanel', () => {
  it('should match snapshot', () => {
    const result = shallow(InfoPanel, {
      propsData: {
        title: '',
      },
    }).element;

    expect(result).toMatchSnapshot();
  });

  it('should have default props and correct types iconName/title', () => {
    const wrapper = shallow(InfoPanel);
    const {
      iconName,
      title,
    } = wrapper.vm.$options.props;

    expect(iconName.type).toEqual(String);
    expect(wrapper.props().iconName).toBe('');

    expect(title.type).toBe(String);
    expect(wrapper.props().title).toBe('');
  });

  it('should render iconName from prop with correct type', () => {
    const wrapper = shallow(InfoPanel, {
      propsData: {
        iconName: 'location',
      },
    });

    expect(wrapper.find(SvgIcon).exists()).toBe(true);
  });

  it('should render title from prop with correct type', () => {
    const title = 'Test title';
    const wrapper = shallow(InfoPanel, {
      propsData: {
        title: 'Test title',
      },
    });

    expect(typeof wrapper.props().title).toBe('string');
    expect(wrapper.props().title).toBe(title);
    expect(wrapper.find('.info-panel__title').text()).toBe(title);
  });

  it('should render any content inside of the slot', () => {
    const wrapper = shallow(InfoPanel, {
      slots: {
        default: ['<p>test</p>'],
      },
    });
    expect(wrapper.find('p').exists()).toBe(true);
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(InfoPanel, {
      propsData: {
        title: 'mock title',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
