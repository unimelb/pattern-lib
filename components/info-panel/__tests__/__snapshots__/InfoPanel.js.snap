// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel should match snapshot 1`] = `
<div
  class="info-panel"
>
  <!---->
   
  <div
    class="info-panel__inner"
  >
    <h4
      class="title--xxs info-panel__title"
    >
      
      
    
    </h4>
     
  </div>
</div>
`;
